conf/gulp-tasks/template/minifytemplatestaskfactory.js

```diff
@@ -0,0 +1,68 @@
+const { src, dest } = require('gulp');
+const uglify = require('gulp-uglify-es').default;
+
+const TemplateType = require('./templatetype');
+const { addLocalePrefix, getFileName } = require('./filenameutils');
+
+/**
+ * MinifyTemplatesTaskFactory is a factory class for creating SDK template
+ * minification tasks.
+ */
+class MinifyTemplatesTaskFactory {
+  constructor (locale) {
+    this._locale = locale;
+  }
+
+  /**
+   * Creates a template minify task for the specified TemplateType,
+   * and customizes the task's name.
+   * @param {TemplateType} templateType
+   * @returns {Function}
+   */
+  create (templateType) {
+    const minifyFunction = this._createMinifyFunction(templateType);
+    Object.defineProperty(minifyFunction, 'name', {
+      value: this._getTaskName(templateType)
+    });
+    return minifyFunction;
+  }
+
+  /**
+   * Returns the minify task for the specified TemplateType.
+   * @param {TemplateType} templateType
+   * @returns {Function}
+   */
+  _createMinifyFunction (templateType) {
+    const fileName = getFileName(templateType, this._locale);
+    return callback => this._minify(callback, fileName);
+  }
+
+  /**
+   * Minifies the designated js file.
+   * @param {Function} callback
+   * @param {string} filename
+   * @returns {stream.Readable}
+   */
+  _minify (callback, filename) {
```

Comment:
Is there any real difference between the minification of the template bundles and the SDK bundles? If not, I think we should look to have a single minify factory. That way, we can consolidate the number of different factories somewhat.

Reply:
sure, could we make an item for that after the js bundle item is done (since they both have a minify factory)? it looks like the logic is almost the same

Reply:
👀 not Oliver but I was looking into this! There are two things that are different, the SDK bundle

1. renames the minified file (the template bundle I think overwrites(?)) 
2. doesn't mangle "ANSWERS" when uglifying

The renaming (1) I think is the larger difference.

Reply:
I think we should look to consolidate if possible. Between all the new factories floating around and the `async` complexities, I think our Gulp architecture is getting sufficiently complicated. I think we should simplify wherever possible. I'm fine if we make an item to address!